soccer/src/soccer/coach/coach_node.cpp

```diff
@@ -0,0 +1,127 @@
+#include "coach_node.hpp"
+
+CoachNode::CoachNode(const rclcpp::NodeOptions& options) : Node("coach_node", options) {
+    coach_state_pub_ =
+        this->create_publisher<rj_msgs::msg::CoachStateInterpretation>("/strategy/coach_state", 10);
+    play_state_change_timer_ =
+        this->create_wall_timer(100ms, [this]() { check_for_play_state_change(); });
+
+    play_state_sub_ = this->create_subscription<rj_msgs::msg::PlayState>(
+        "/referee/play_state", 10,
+        [this](const rj_msgs::msg::PlayState::SharedPtr msg) { play_state_callback(msg); });
+
+    world_state_sub_ = this->create_subscription<rj_msgs::msg::WorldState>(
+        "/vision_filter/world_state", 10,
+        [this](const rj_msgs::msg::WorldState::SharedPtr msg) { world_state_callback(msg); });
+
+    // initialize all of the robot status subscriptions
+    // initialize our robots
+    for (int i = 0; i < kRobotsPerTeam; i++) {
```

Comment:
can we change this to kNumShells? just in case we use the shells for robots >= 6

Reply:
not certain that's the exact syntax, btw, but it's in the same file as kRobotsPerTeam